server/controllers/chartsController/helpers.js

```diff
@@ -303,3 +331,67 @@ export const calculateSubjectVariableDayCount = (
 
   return subjectsDayDataAndVariable.length
 }
+
+export const buildFacetForFilters = ({
+  isSexAtBirthFilterActive,
+  isInclusionCriteriaFilterActive,
+  userAccess,
+}) => {
+  const facetForFilters = {}
+
+  if (isSexAtBirthFilterActive) {
+    facetForFilters.socioDemographics = [
+      {
+        $match: {
+          assessment: SOCIODEMOGRAPHICS_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+          socioDemographicsCollection: MONGO_COLLECTION_STRING,
+        },
+      },
+    ]
+  }
+  if (isInclusionCriteriaFilterActive) {
+    facetForFilters.inclusionCriteria = [
+      {
+        $match: {
+          assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+          inclusionCriteriaCollection: MONGO_COLLECTION_STRING,
+        },
+      },
+    ]
+  }
+
+  return facetForFilters
+}
+
+export const intersectSubjectsFromFilters = (
+  filters,
+  mapForResultsFromFilters
+) => {
+  const [filterListA, filterListB] = Object.values(filters).map((filter) => {
```

Comment:
what if there are more than 2 filters?

Reply:
If there are more than 2 filters then we add a third list to the constant and process it through the logic of the map.
```suggestion
  const [filterListA, filterListB, filterListC] = Object.values(filters).map((filter) => {
```

Reply:
we could also build this where we don't need to add that additional filter. Either using `const filters = ...` or `const [firstFilter, ...rest] = ...`. Then we don't need to remember to add more logic here when another filter is added.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/280/commits/7c4ba3e249a5a05624e8ee92b34c5d42aa3aab7e

Reply:
also changed the Map to an object
https://github.com/AMP-SCZ/dpdash/pull/280/commits/3624746805fe17d9c3bf8f10cdfa7ff572e8557e